apps/web/src/components/forms/token.tsx

```diff
@@ -0,0 +1,208 @@
+'use client';
+
+import { useState } from 'react';
+
+import { useRouter } from 'next/navigation';
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Loader } from 'lucide-react';
+import { useForm } from 'react-hook-form';
+import type { z } from 'zod';
+
+import { useCopyToClipboard } from '@documenso/lib/client-only/hooks/use-copy-to-clipboard';
+import { TRPCClientError } from '@documenso/trpc/client';
+import { trpc } from '@documenso/trpc/react';
+import { ZCreateTokenMutationSchema } from '@documenso/trpc/server/api-token-router/schema';
+import { cn } from '@documenso/ui/lib/utils';
+import { Button } from '@documenso/ui/primitives/button';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@documenso/ui/primitives/form/form';
+import { Input } from '@documenso/ui/primitives/input';
+import { useToast } from '@documenso/ui/primitives/use-toast';
+
+import DeleteTokenDialog from '~/components/(dashboard)/settings/token/delete-token-dialog';
+
+export type ApiTokenFormProps = {
+  className?: string;
+};
+
+type TCreateTokenMutationSchema = z.infer<typeof ZCreateTokenMutationSchema>;
+
+export const ApiTokenForm = ({ className }: ApiTokenFormProps) => {
+  const router = useRouter();
+  const [, copy] = useCopyToClipboard();
+  const { toast } = useToast();
+  const [newlyCreatedToken, setNewlyCreatedToken] = useState({ id: 0, token: '' });
+  const [showNewToken, setShowNewToken] = useState(false);
+
+  const { data: tokens, isLoading: isTokensLoading } = trpc.apiToken.getTokens.useQuery();
+  const { mutateAsync: createTokenMutation } = trpc.apiToken.createToken.useMutation({
+    onSuccess(data) {
+      setNewlyCreatedToken({ id: data.id, token: data.token });
+    },
+  });
+
+  const form = useForm<TCreateTokenMutationSchema>({
+    resolver: zodResolver(ZCreateTokenMutationSchema),
+    values: {
+      tokenName: '',
+    },
+  });
+
+  const onDelete = (tokenId: number) => {
```

Comment:
This method is called in "delete-token-dialog.tsx" after a successful mutation to avoid deleting the snippet with the newly created token from the screen when users delete any of their tokens except the newly created one.

Not sure if it's the best approach though.